src/lib.rs

```diff
@@ -726,4 +738,32 @@ mod tests {
 
         assert!(total > 5 && total < 15);
     }
+
+    /// See: https://github.com/udoprog/leaky-bucket/issues/5#issuecomment-703205787
+    #[tokio::test]
+    async fn test_graceful_shutdown_coordinator() {
+        let interval = Duration::from_millis(20);
+
+        let mut buckets = LeakyBuckets::new();
+
+        let leaky = buckets
+            .rate_limiter()
+            .tokens(0)
+            .max(10)
+            .refill_amount(1)
+            .refill_interval(interval)
+            .build()
+            .expect("build rate limiter");
+
+        let coordinator = buckets.coordinate().unwrap();
+
+        let task = tokio::spawn(async move {
+            assert!(coordinator.await.is_ok());
```

Comment:
np: `timeout(Duration::from_secs(1), coordinator.await).expect("must not timeout").expect("must be ok")` to avoid hanging tests if the functionality is broken.

Reply:
The test runner has a built-in timeout of 60 seconds per test. It will hang sure, but luckily not forever. So as a principle I generally don't bother with adding timeouts unless it's needed for the test itself.